--- gstnvbaseenc.old	2021-10-27 19:32:43.438789871 +0300
+++ gstnvbaseenc.c	2021-12-15 14:48:04.922995069 +0300
@@ -638,6 +638,11 @@
     return FALSE;
   }
 
+  if (nvenc->items == 0x0) {
+    GST_WARNING_OBJECT (nvenc, "Items empty!");
+    return FALSE;
+  }
+
   features = gst_caps_get_features (caps, 0);
 #if HAVE_NVCODEC_GST_GL
   if (features && gst_caps_features_contains (features,
@@ -2333,7 +2338,7 @@
   pic_params.frameIdx = frame->system_frame_number;
 
   if (GST_VIDEO_CODEC_FRAME_IS_FORCE_KEYFRAME (frame))
-    pic_params.encodePicFlags = NV_ENC_PIC_FLAG_FORCEIDR;
+    pic_params.encodePicFlags = NV_ENC_PIC_FLAG_FORCEIDR | NV_ENC_PIC_FLAG_OUTPUT_SPSPPS;
   else
     pic_params.encodePicFlags = 0;
 
